repo/blob/s3/s3_storage_config.go

```diff
@@ -15,8 +16,9 @@ const ConfigName = ".storageconfig"
 
 // PrefixAndStorageClass defines the storage class to use for a particular blob ID prefix.
 type PrefixAndStorageClass struct {
-	Prefix       blob.ID `json:"prefix"`
-	StorageClass string  `json:"storageClass"`
+	Prefix          blob.ID   `json:"prefix"`
+	StorageClass    string    `json:"storageClass"`
+	RetainUntilDate time.Time `json:"retainedUntilDate"`
```

Comment:
See comment above about putting the default retention period in the `kopia.repository` blob instead.